packages/panzer/adapters-stk/test/evaluator_tests/tpetra_blocked_scatter_dirichlet_residual.cpp

```diff
@@ -0,0 +1,1145 @@
+// @HEADER
+// *****************************************************************************
+//           Panzer: A partial differential equation assembly
+//       engine for strongly coupled complex multiphysics systems
+//
+// Copyright 2011 NTESS and the Panzer contributors.
+// SPDX-License-Identifier: BSD-3-Clause
+// *****************************************************************************
+// @HEADER
+
+#include <Teuchos_ConfigDefs.hpp>
+#include <Teuchos_UnitTestHarness.hpp>
+#include <Teuchos_RCP.hpp>
+#include <Teuchos_TimeMonitor.hpp>
+
+using Teuchos::RCP;
+using Teuchos::rcp;
+
+#include "Teuchos_DefaultComm.hpp"
+#include "Teuchos_GlobalMPISession.hpp"
+
+#include "Panzer_FieldManagerBuilder.hpp"
+#include "Panzer_BlockedDOFManager.hpp"
+#include "Panzer_BlockedTpetraLinearObjFactory.hpp"
+#include "Panzer_PureBasis.hpp"
+#include "Panzer_BasisIRLayout.hpp"
+#include "Panzer_Workset.hpp"
+#include "Panzer_GatherOrientation.hpp"
+#include "Panzer_ScatterDirichletResidual_BlockedTpetra.hpp"
+#include "Panzer_GatherSolution_BlockedTpetra.hpp"
+#include "Panzer_LOCPair_GlobalEvaluationData.hpp"
+#include "Panzer_GlobalEvaluationDataContainer.hpp"
+#include "Panzer_ParameterList_GlobalEvaluationData.hpp"
+
+#include "Panzer_STK_Version.hpp"
+#include "PanzerAdaptersSTK_config.hpp"
+#include "Panzer_STK_Interface.hpp"
+#include "Panzer_STK_SquareQuadMeshFactory.hpp"
+#include "Panzer_STK_SetupUtilities.hpp"
+#include "Panzer_STKConnManager.hpp"
+
+#include "Teuchos_DefaultMpiComm.hpp"
+#include "Teuchos_OpaqueWrapper.hpp"
+
+#include "Thyra_VectorStdOps.hpp"
+#include "Thyra_ProductVectorBase.hpp"
+#include "Thyra_SpmdVectorBase.hpp"
+
+#include "Tpetra_CrsMatrix.hpp"
+#include "Tpetra_Map.hpp"
+
+#include "user_app_EquationSetFactory.hpp"
+
+#include <cstdio> // for get char
+#include <vector>
+#include <string>
+
+#include "Panzer_Evaluator_WithBaseImpl.hpp"
+
+namespace panzer
+{
+   typedef Teuchos::ArrayRCP<const double>::size_type size_type;
+
+   using TpetraBlockedLinObjFactoryType = panzer::BlockedTpetraLinearObjFactory<panzer::Traits, double, panzer::LocalOrdinal, panzer::GlobalOrdinal>;
+   using TpetraBlockedLinObjContainerType = panzer::BlockedTpetraLinearObjContainer<double, panzer::LocalOrdinal, panzer::GlobalOrdinal>;
+   using Tpetra_CrsMatrix = Tpetra::CrsMatrix<double, panzer::LocalOrdinal, panzer::GlobalOrdinal>;
+   using Thyra_TpetraLinearOp = Thyra::TpetraLinearOp<double, LocalOrdinal, GlobalOrdinal>;
+
+   Teuchos::RCP<panzer::PureBasis> buildBasis(std::size_t worksetSize, const std::string &basisName);
+   void testInitialization(const Teuchos::RCP<Teuchos::ParameterList> &ipb);
+   Teuchos::RCP<panzer_stk::STK_Interface> buildMesh(int elemX, int elemY);
+
+   TEUCHOS_UNIT_TEST(block_assembly, scatter_dirichlet_residual)
+   {
+
+#ifdef HAVE_MPI
```

Comment:
can probably just remove the ifdefs. MPI is a required TPL dependency for panzer so the else can't happen. I think this was leftover from the early days when it did support a serial build.